src/app/forms/custom-sections/letters/letters.component.ts

```diff
@@ -73,20 +76,35 @@ export class LettersComponent implements OnInit, OnDestroy, OnChanges {
       }
       : undefined;
   }
+  // checking if the Technical Record History has current status code.
+  get checkForCurrentRecordInHistory(): boolean {
+    let hasCurrent = false;
+    this.techRecordService.techRecordHistory$.subscribe((historyArray: TechRecordSearchSchema[] | undefined) => {
+      historyArray?.forEach((history: TechRecordSearchSchema) => {
+        if (history.techRecord_statusCode === StatusCodes.CURRENT
+          && this.techRecord?.techRecord_statusCode === StatusCodes.PROVISIONAL) hasCurrent = true;
+      });
+    });
+    return hasCurrent;
+  }
```

Comment:
All the code in this PR works functionality, and I'm happy with it, but this bit causes a memory leak (can show you how if needed). Not sure what standard practice in VTM is @shivangidas but this looks like it could be its own selector on the tech record history service that you:
a. Subscribe to on ngOnInit
b. Have a takeUntil destroyed pipe, so it stops listening after this component is destroyed (stopping the memory leak)
c. assign as a member of the class so it doesn't create a new subscription on every component re-render, as it currently does using the getter approach.

Reply:
Currently, there's a bit of everything, but would certainly like the solution recommended, think there already is a tech record history selector present `selectTechRecordHistory` as well